x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2777 @@
+
+will perform migration but does not allow for customization during the duplication process.
+
+So we can analyze the expected outputs of the migration of all these 8 endpoints together.
+
+The resulting values for `immutable` and `rule_source` when calling these endpoints, and the migration being performed in the background, should be as follows:
+
+<table>
+  <thead>
+    <tr>
+      <th>Migration use case</th>
+      <th>Current value of <pre>rule_source</pre></th>
+      <th>Current value of <pre>immutable</pre></th>
+      <th>Any field diverges <br> from base version after update?</th>
+      <th>Results</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td><b>Custom rule</b> (not migrated yet)</td>
+      <td>
+        <pre>undefined</pre>
+      </td>
+      <td><pre>false</pre></td>
+      <td>N/A - Doesn't apply for custom rules</td>
+      <td>
+        <pre>
+{
+  rule_source: {
+    type: 'internal'
+  },
+  immutable: false,
+}
+        </pre>
+      </td>
+    </tr>
+    <tr>
+      <td><b>Custom rule</b> (already migrated)</td>
+      <td>
+        <pre>
+{
+  type: 'internal'
+}
+        </pre>
+      </td>
+      <td><pre>false</pre></td>
+      <td>N/A - Doesn't apply for custom rules</td>
+      <td>
+        <pre>
+{
+  rule_source: {
+    type: 'internal'
+  },
+  immutable: false,
+}
+        </pre>
+      </td>
+    </tr>
+    <tr>
+      <td><b>Prebuilt rule</b> (not yet migrated, no customizations)</td>
+      <td>
+        <pre>undefined</pre>
+      </td>
+      <td><pre>true</pre></td>
+      <td>No</td>
+      <td>
+        <pre>
+{
+  rule_source: {
+    type: 'external',
+    isCustomized: false,
+    ...
+  },
+  immutable: true,
+}
+        </pre>
+      </td>
+    </tr>
+    <tr>
+      <td><b>Prebuilt rule</b> (not yet migrated, with customizations)</td>
+      <td>
+        <pre>undefined</pre>
+      </td>
+      <td><pre>true</pre></td>
+      <td>Yes</td>
+      <td>
+        <pre>
+{
+  rule_source: {
+    type: 'external',
+    isCustomized: true,
+    ...
+  },
+  immutable: true,
+}
+        </pre>
+      </td>
+    </tr>
+    <tr>
+      <td><b>Prebuilt rule</b> (already migrated, no customizations)</td>
+      <td>
+        <pre> 
+{
+  type: 'external',
+  isCustomized: false,
+    ...
+}
+          </pre>
+      </td>
+      <td><pre>true</pre></td>
+      <td>No</td>
+      <td>
+        <pre>
+{
+  rule_source: {
+    type: 'external',
+    isCustomized: false,
+    ...
+  },
+  immutable: true,
+}
+        </pre>
+      </td>
+    </tr>
+    <tr>
+      <td><b>Prebuilt rule</b> (already migrated, with customizations)</td>
+      <td>
+        <pre> 
+{
+  type: 'external',
+  isCustomized: true,
+    ...
+}
+          </pre>
+      </td>
+      <td><pre>true</pre></td>
+      <td>Yes</td>
+      <td>
+        <pre>
+{
+  rule_source: {
+    type: 'external',
+    isCustomized: true,
+    ...
+  },
+  immutable: true,
+}
+        </pre>
+      </td>
+    </tr>
```

Comment:
The currently stored value of rule source should be this, I believe:

```js
{
  type: 'external',
  isCustomized: false,
    ...
}
```

Reply:
In this case, both examples work: if the rules is or  not customized and during the editing it gets customized, then isCustomized should be `true` in both cases.

But this makes me think if we should add one more case: if the initial value of `isCustomized` is `true` and the editing of the rule makes it match the `base` version, then the resulting `isCustomized` should be `false`. But maybe the detail is not needed in the RFC

Reply:
- Add scenario to table.